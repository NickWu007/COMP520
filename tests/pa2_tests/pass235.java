//PA2 fail qualified reference expression
class A {
    private void p(){
        int a = x.y[4];
    }
}
