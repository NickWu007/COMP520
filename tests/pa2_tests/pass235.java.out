======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (private) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     INT BaseType
  .   .   .     "a" varname
  .   .   .   RefExpr
  .   .   .     IxQRef
  .   .   .       "y" Identifier
  .   .   .       LiteralExpr
  .   .   .         "4" IntLiteral
  .   .   .       IdRef
  .   .   .         "x" Identifier
=============================================
